scripts/feature_tests/upgrade/upgrade_common.sh

```diff
@@ -237,23 +234,26 @@ function worker_has_correct_replicas() {
       echo "Waiting for all replicas of worker nodes to join the cluster"
     fi
 
-    for i in {1..3600}; do
+    for i in {1..1800}; do
+        wr_replicas=$(kubectl get bmh -n metal3 | grep -i provisioned | grep -c worker)
         if [[ "${replicas}" -eq 0 ]]; then
-            wr_replicas=$(kubectl get bmh -n metal3 | grep -c worker)
-        else
-            kubectl get secrets "${CLUSTER_NAME}"-kubeconfig -n "${NAMESPACE}" -o json | \
-            jq -r '.data.value'| base64 -d > /tmp/kubeconfig-"${CLUSTER_NAME}".yaml
-            wr_replicas=$(kubectl --kubeconfig=/tmp/kubeconfig-"${CLUSTER_NAME}".yaml get nodes |
-                awk 'NR>1' | grep -vc master)
```

Comment:
same thing here

Reply:
When scaling to zero, the target cluster worker node deletion has failed very rarely. That's why the scale to zero condition is kept separate.
When scaling to >=1, the management cluster status is checked first to avoid 100+ times writing the kubeconfig to file. Target cluster check is done after the management cluster check is met.